### PR TITLE
[JAVA]WeirdSortingArnab.java

### PR DESCRIPTION
**Pull request by codebook-2000**
This [code](https://github.com/dscnsec/DSC-NSEC-Algorithms/commit/c6cf5e7b8dfcbb773ab13ee071a784395a6818d7) attains the following features:-
- [X] Code is added in right directory(Folder).
- [X] Brief Description of Solution is given.
- [X] Time and Space Complexity is mentioned
- [X] Code is well indented.
- [X] Comments are clearly written wherever necessary
- [X] Code is successfully compiled and passes all test cases.

### CODE REVIEW

**Review by OverLordAct — COMMENTED** [review]
Do you think you can do this in a single pass instead of counting?

File: 1. Array/weird_sorting/WeirdSortingArnab.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import java.io.*;
+
+
+class Weird_Sorting
+{
+    static BufferedReader buf=new BufferedReader(new InputStreamReader(System.in));
+    static int arr[];
+    static void solve() throws java.lang.Exception
+    {
+        int n=Integer.parseInt(buf.readLine());
```
Comment (OverLordAct) [inline 1. Array/weird_sorting/WeirdSortingArnab.java]: Take input from main() method into an array and pass it as an argument into solve() method

**Comment by codebook-2000** [comment]
> Do you think you can do this in a single pass instead of counting?

In a single pass, if we do then we require sorting. But Time complexity mentioned is O(n). That's why I did it in this manner

File: 1. Array/weird_sorting/WeirdSortingArnab.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ * @author [codebook-2000] (https://github.com/codebook-2000)
+ */
+ 
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+
+class Weird_Sorting
+{
+    static int arr[];
```
Comment (OverLordAct) [inline 1. Array/weird_sorting/WeirdSortingArnab.java]: No need to create a static arr[] here. You can directly pass the reference of the array to solve() and use it later.

File: 1. Array/weird_sorting/Weird_Sorting_Arnab.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+ * Time Complexity-O(n) Extra Space Complexity-O(1)
+ * @author [codebook-2000] (https://github.com/codebook-2000)
+ */
+ 
+
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+
+class Weird_Sorting_Arnab
```
Comment (OverLordAct) [inline 1. Array/weird_sorting/Weird_Sorting_Arnab.java]: Please change the name of the class and file as per Java convention

Comment (codebook-2000) [reply 1. Array/weird_sorting/Weird_Sorting_Arnab.java]: <h2>I have updated it, Kindly check it</h2>

**Pull request merged**